src/config.rs

```diff
@@ -0,0 +1,193 @@
+use serde::Deserialize;
+use std::env;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::PathBuf;
+
+/// Represents the configuration loaded from the `pyproject.toml` file.
+///
+/// The struct is created using the `from_file()` or `load()` functions. It has a `tool` field
+/// that contains an optional `ToolConfig` instance.
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    #[serde(rename = "tool")]
+    pub tool: Option<ToolConfig>,
+}
+
+/// Represents the `tool` configuration section in the `pyproject.toml` file.
+///
+/// The struct is used in the `Config` struct to deserialize the `tool` section. It has an
+/// optional `black` field that contains a `BlackConfig` instance.
+#[derive(Debug, Deserialize)]
+pub struct ToolConfig {
+    pub black: Option<BlackConfig>,
+}
+
+/// Represents the `black` configuration section in the `pyproject.toml` file.
+///
+/// The struct is used in the `ToolConfig` struct to deserialize the `black` section. It has
+/// an optional `line_length` field that specifies the maximum line length, and an optional
+/// `target_version` field that specifies the target Python versions.
+#[derive(Debug, Deserialize)]
+pub struct BlackConfig {
+    #[serde(rename = "line-length")]
+    pub line_length: Option<i32>,
+    #[serde(rename = "target-version")]
+    pub target_version: Option<Vec<String>>,
+}
+
+impl Config {
+    /// Loads the configuration from the `pyproject.toml` file.
+    ///
+    /// The function searches for the `pyproject.toml` file in the current directory and its
+    /// parent directories. It returns a `Config` instance or a default instance if the file
+    /// is not found.
+    ///
+    /// # Arguments
+    ///
+    /// * `target_file_name`: An optional `&str` that specifies the target file name. The
+    /// default value is `"pyproject.toml"`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let config = Config::load(None);
+    /// println!("{:?}", config);
+    /// ```
+    pub fn load(target_file_name: Option<&str>) -> Self {
+        let mut path = env::current_dir().unwrap();
+        let target_file_name = target_file_name.unwrap_or("pyproject.toml");
+
+        while path.pop() {
```

Comment:
This will pop current dir before check.

For example, I have config in my current dir `/project/name/pyproject.toml`

In this case first check will be on `/project/pyproject.toml`

This works ok

```rust
        loop {
            let potential_config_path = path.join(target_file_name);
            if potential_config_path.is_file() {
                return Config::from_file(potential_config_path)
                    .expect("Failed to load config file");
            }
            match path.pop() {
                true => (),
                false => return Config { tool: None },
            }
        }
```